Mathlib/Init/WF.lean

```diff
@@ -0,0 +1,12 @@
+
+noncomputable def skipLeft {β : α → Sort _}
```

Comment:
They are annoying to use if they are namespaced. Hopefully leanprover/lean4#879 will be handled at some point, which will make this unnecessary in most cases.

There is a similar definition in `Init.WF`, but it is nondependent. Here the idea is to introduce the parameter to skip as a variable in the context so it can be used in later relations (e.g. in `Fin.upRel`).

Reply:
> They are annoying to use if they are namespaced.

Really?  `open PSigma` works just fine (for the core function).  (I completely agree that leanprover/lean4#849 is the proper fix here though.)

> There is a similar definition in Init.WF, but it is nondependent.

I am not sure if this makes it better; we now have two functions with the same name, same purpose, but different types.  At least a prime would be good here for disambiguation.

Reply:
It wouldn't be good to have `open PSigma` at the top level because the user probably isn't doing anything related to `PSigma` except for the specification of the well founded relation (they didn't ask for it, it is just how the compilation works). If we really want to put these in the PSigma namespace then `termination_by` should automatically open this namespace (or perhaps some other namespace dedicated to termination relations), because it is so different from everything else in the user's file.